src/client/content_mapblock.cpp

```diff
@@ -1664,48 +1663,51 @@ void MapblockMeshGenerator::drawMeshNode()
 
 	if (cur_node.f->mesh_ptr) {
 		// clone and rotate mesh
-		private_mesh = true;
 		mesh = cloneMesh(cur_node.f->mesh_ptr);
+		bool modified = true;
 		if (facedir)
 			rotateMeshBy6dFacedir(mesh, facedir);
 		else if (degrotate)
 			rotateMeshXZby(mesh, 1.5f * degrotate);
-		recalculateBoundingBox(mesh);
-		meshmanip->recalculateNormals(mesh, true, false);
+		else
+			modified = false;
+		if (modified) {
+			recalculateBoundingBox(mesh);
+			// FIXME: we should rotate the normals too, instead of recalculating
```

Comment:
Why leave this as a `FIXME`? Isn't this pretty easy to fix or am I missing something:

```diff
diff --git a/src/client/mesh.cpp b/src/client/mesh.cpp
index 3f947ae50..06b9f1d67 100644
--- a/src/client/mesh.cpp
+++ b/src/client/mesh.cpp
@@ -249,10 +249,14 @@ static void rotateMesh(scene::IMesh *mesh, float degrees)
        float c = std::cos(degrees);
        float s = std::sin(degrees);
        auto rotator = [c, s] (video::S3DVertex *vertex) {
-               float u = vertex->Pos.*U;
-               float v = vertex->Pos.*V;
-               vertex->Pos.*U = c * u - s * v;
-               vertex->Pos.*V = s * u + c * v;
+               auto rotate_vec = [c, s] (v3f &vec) {
+                       float u = vec.*U;
+                       float v = vec.*V;
+                       vec.*U = c * u - s * v;
+                       vec.*V = s * u + c * v;
+               };
+               rotate_vec(vertex->Pos);
+               rotate_vec(vertex->Normal);
        };
        applyToMesh(mesh, rotator);
 }
```

Reply:
looks fine but I didn't want it to be me who figures it out

Reply:
https://github.com/minetest/minetest/pull/15523/commits/d0b886bd2750ac23ca7887db210acb74a0f45d12